writer.go

```diff
@@ -99,27 +153,35 @@ func newWriterDictLevel(w io.Writer, cd *CDict, compressionLevel int) *Writer {
 }
 
 // Reset resets zw to write to w using the given dictionary cd and the given
-// compressionLevel.
+// compressionLevel. Other parameters set via WriterParams remain unchanged.
 func (zw *Writer) Reset(w io.Writer, cd *CDict, compressionLevel int) {
 	zw.inBuf.size = 0
 	zw.inBuf.pos = 0
 	zw.outBuf.size = cstreamOutBufSize
 	zw.outBuf.pos = 0
 
 	zw.cd = cd
-	initCStream(zw.cs, zw.cd, compressionLevel)
+	params := &WriterParams{
```

Comment:
Could you verify whether this line leads to memory allocation? If yes, then I'd suggest passing `params` by value to `initCStream` in order to remove this memory allocation, since `Writer.Reset` is usually used in functions sensitive for performance, which want minimizing memory allocations.

Reply:
in `Reset` actually it doesn't.
But as you suggested I added `ResetWriterParams` method and this line does allocate even when passing params to `initCStream`, but I'm not sure there's an easy way around that since, as you can see, `params` escapes to heap because of the of the assignment of `params.Dict` `to the receveir's `cd`, namely  `zw.cd`.

```
./writer.go:174:50: parameter params leaks to {heap} with derefs=1:
./writer.go:174:50:   flow: {heap} = *params:
./writer.go:174:50:     from params.Dict (dot of pointer) at ./writer.go:180:16
./writer.go:174:50:     from zw.cd = params.Dict (assign) at ./writer.go:180:8
```
Passing `params` by value or pointer to `initCStream` doesn't change that fact.

Let me know if you see something else we could try to avoid that allocation